lib/tasks/allrecipe-category-tags.rake

```diff
@@ -10,23 +10,27 @@ def get_category_info(main_category_name, main_category_url)
 
   @tries = 0
   begin
-    puts main_category_name
     puts main_category_url
     browser = open(main_category_url).read
     html_doc = Nokogiri::HTML(browser)
-
+    
     sub_category_elements = html_doc.css('a#hlSubNavItem')
+    if html_doc.css('ul#subNavGroupContainer.cat.related li') != nil
+      element_to_remove = (html_doc.css('ul#subNavGroupContainer.cat.related li').size.to_i + 1) * -1
```

Comment:
Too much calculations. Should be easier to understand. Use `[0..(total_length - to_remove_length)]`